### PR TITLE
Testing: Husky, Prisma-seed, GitHub templates

### PR DESCRIPTION
**Pull request by IAmRiteshKoushik**
> This PR is still a WIP. Do not merge. 

This PR introduces the following:
- [X] `.husky/` for pre-commit hooks (needed to maintain single lock file)
- [x] CI-pipeline added
- [x] Issue Template in `.github/` 
- [X] PR Template in `.github/`
- [ ] Prisma `seed.ts` script

Review and merge @Ashrockzzz2003

### CODE REVIEW

**Comment by IAmRiteshKoushik** [comment]
@Ashrockzzz2003 Would it be a good idea to introduce CI to check the build before merging the PRs ?

**Comment by Ashrockzzz2003** [comment]
> @Ashrockzzz2003 Would it be a good idea to introduce CI to check the build before merging the PRs ?

Yep 
It's better to have to ensure lesser trouble in deployment.

File: .github/ISSUE_TEMPLATE/bug.md
Comment (Ashrockzzz2003) [inline .github/ISSUE_TEMPLATE/bug.md:1]: Add about linking correct labels to the issue and rules for who to assign to.

File: .github/ISSUE_TEMPLATE/chore.md
Comment (Ashrockzzz2003) [inline .github/ISSUE_TEMPLATE/chore.md:1]: Include about listing tasks and subtasks with this format 

```md
- [ ] task
```

File: .github/ISSUE_TEMPLATE/feature.md
Comment (Ashrockzzz2003) [inline .github/ISSUE_TEMPLATE/feature.md:1]: Probably mention about adding the tasks it would involve too if possible.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Ashrockzzz2003) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: How about a TL;DR for every PR on top for quick identification.

**Review by Ashrockzzz2003 — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (IAmRiteshKoushik) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: A simple description should suffice. The project is not expecting a huge inflow of contributors. This template is added only to make things easier for @Nandgopal-R @adarsshkg-codebank and the upcoming maintainers of this project. Simple description should suffice.

File: .github/ISSUE_TEMPLATE/feature.md
Comment (IAmRiteshKoushik) [reply .github/ISSUE_TEMPLATE/feature.md:1]: Nice one. Have added this in the latest commit.

File: .github/ISSUE_TEMPLATE/chore.md
Comment (IAmRiteshKoushik) [reply .github/ISSUE_TEMPLATE/chore.md:1]: This change is added in the latest commit where a person creating an issue gets the option to add subtasks which might associated so that the contributors and maintainers get better clarity on the things to address.

File: .github/ISSUE_TEMPLATE/bug.md
Comment (IAmRiteshKoushik) [reply .github/ISSUE_TEMPLATE/bug.md:1]: Label linking will always remain fairly subjective. Again, not expecting huge inflow of contributors. Will throw in a tip or two.

**Comment by IAmRiteshKoushik** [comment]
> > @Ashrockzzz2003 Would it be a good idea to introduce CI to check the build before merging the PRs ?
> 
> Yep It's better to have to ensure lesser trouble in deployment.

Can you open a PR and help out with this. Have already added the `workflows` directory

**Review by Nandgopal-R — APPROVED** [review]

**Review by Nandgopal-R — APPROVED** [review]

**Pull request merged**
